### PR TITLE
Migrate to PDK

### PR DESCRIPTION
**Pull request by albertvaka**
Puppet Development Kit (PDK) is the recommended way to build and release Puppet modules.

PDK adds a base set of files to the projects (think skel in the Unix world), which I've tried to respect and merge with the stuff we already had.

Gem dependencies are now grouped under the meta-packages `puppet-module-*`, although unfortunately this is not available for Ruby 2.2 and earlier, where we still have to list dependencies manually. Also, these meta-packages don't work with unsupported combinations like Ruby 2.6 with Puppet < 6, so I had to remove those from the test matrix.

I haven't enabled the rubocop check since it has lots of offenses, but will do so on a separate PR.

Releases of the module can now be done using `pdk build`.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+Style/WordArray:
+  EnforcedStyle: brackets
+Style/CollectionMethods:
+  Enabled: true
+Style/MethodCalledOnDoEndBlock:
+  Enabled: true
+Style/StringMethods:
+  Enabled: true
+GetText/DecorateFunctionMessage:
+  Enabled: false
```
Comment (truthbk) [inline .rubocop.yml]: we've disabled a bunch of stuff here? Are all these necessary? Definitely fine if it is, just wondering.

File: Gemfile.lock [hunk 2]
```diff
@@ -1,129 +1,218 @@
 GEM
   remote: https://rubygems.org/
   specs:
+    addressable (2.7.0)
```
Comment (truthbk) [inline Gemfile.lock]: We should probably state somewhere what ruby, puppet version was used to generate the lock file; I guess we could call that the reference platform.

Like we discussed last time, there are so many combinations this was probably why it hadn't been included in the repo in the past.

File: manifests/init.pp [hunk 3]
```diff
@@ -426,7 +426,7 @@
         msi_location        => $win_msi_location,
         api_key             => $api_key,
         hostname            => $host,
-        service_name        => $service_name,
+        service_name        => $datadog_agent::params::service_name,
```
Comment (truthbk) [inline manifests/init.pp]: 🍰

File: .rubocop.yml (hunk 1 above)
Comment (albertvaka) [reply .rubocop.yml]: PDK adds this file as-is by default. We aren't using it yet, though.

File: Gemfile.lock (hunk 2 above)
Comment (albertvaka) [reply Gemfile.lock]: There is a default in the Gemfile that is used unless a specific version is given. This Gemfile.lock is generated from that default.

**Pull request merged**